### PR TITLE
codec(ticdc): simple decoder set the table id on handle key only event and make consumer more strict on the ts order

### LINKED ISSUE #11088: flaky test: simple protocol handle key only may failed
### Which jobs are flaking?

kafka integration test

### Which test(s) are flaking?

simple protocol avro handle key only

### Jenkins logs or GitHub Actions link

```log
(paste logs here)
```

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #11088

### What is changed and how it works?

* set the table id when decode the handle key only event
* consumer panics if commit-ts less than the resolved ts
* update resolved ts after receive resolved ts event and ddl event.

#### Tests 

 - Integration test

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Review by CharlesCheung96 — APPROVED** [review]

File: cmd/kafka-consumer/main.go [hunk 1]
```diff
@@ -669,43 +674,42 @@ func (c *Consumer) ConsumeClaim(session sarama.ConsumerGroupSession, claim saram
...
+				tableID := row.PhysicalTableID
+				// simple protocol decoder should have set the table id already.
+				if c.option.protocol != config.ProtocolSimple {
+					var partitionID int64
+					if row.TableInfo.IsPartitionTable() {
+						partitionID = row.PhysicalTableID
+					}
+					tableID = c.fakeTableIDGenerator.
+						generateFakeTableID(row.TableInfo.GetSchemaName(), row.TableInfo.GetTableName(), partitionID)
+					row.TableInfo.TableName.TableID = tableID
```
Comment (asddongmen) [inline cmd/kafka-consumer/main.go:704]: Why is this needed?

Comment (3AceShowHand) [reply cmd/kafka-consumer/main.go:704]: to make other protocol, such as canal-json works, they does not encode the table id into the message

Comment (asddongmen) [reply cmd/kafka-consumer/main.go:704]: Would it still overwrite the actual value of `TableInfo.TableName.TableID` as we saw in the log of failed tests?

Comment (3AceShowHand) [reply cmd/kafka-consumer/main.go:704]: simple protocol won't hit this code snippets

**Review by asddongmen — APPROVED** [review]

File: cmd/kafka-consumer/main.go (hunk 1 above)
Comment (asddongmen) [reply cmd/kafka-consumer/main.go:704]: Got it.

**Pull request merged**